docker-compose.yml

```diff
@@ -0,0 +1,21 @@
+version: '3.5'
+
+services:
+  contracts-env:
+    security_opt:
+      - no-new-privileges
+    user: 1001:1001
+    env_file:
+      - .env
+    build:
```

Comment:
I think you can drop the Dockerfile, avoid building altogether, use `image: node:16` here, and just mount the solidity repo root. Hardhat takes care of installing the proper Solidity compiler for you.

Reply:
Good call!